### PR TITLE
Fixing combobox styles for quiet and invalid

### PR DESCRIPTION
**Pull request by LFDanLu**
Closes https://github.com/adobe/react-spectrum/issues/1980

## ✅ Pull Request Checklist:

- [ ] Included link to corresponding [React Spectrum GitHub Issue](https://github.com/adobe/react-spectrum/issues).
- [ ] Added/updated unit tests and storybook for this change (for new code or code which already has tests).
- [ ] Filled out test instructions.
- [ ] Updated documentation (if it already exists for this component).
- [ ] Looked at the Accessibility Practices for this feature - [Aria Practices](https://www.w3.org/TR/wai-aria-practices-1.1/)

## 📝 Test Instructions:

look at the chromatic stories for combobox: https://www.chromatic.com/build?appId=5f0dd5ad2b5fc10022a2e320&number=78

## 🧢 Your Project:

RSP

### CODE REVIEW

File: packages/@adobe/spectrum-css-temp/components/inputgroup/index.css [hunk 1]
```diff
@@ -44,7 +44,8 @@ governing permissions and limitations under the License.
     flex-shrink: 0;
   }
 
-  &.is-disabled {
+  /* Quiet or invalid inputgroup field button should always have a border width. */
+  &.is-disabled:not(.spectrum-InputGroup--invalid):not(.spectrum-InputGroup--quiet) {
```
Comment (snowystinger) [inline packages/@adobe/spectrum-css-temp/components/inputgroup/index.css:48]: have you run chromatic on this branch? i recall doing something like this that broke something else, i don't recall what it was, but i think it came up in chromatic

Comment (LFDanLu) [reply packages/@adobe/spectrum-css-temp/components/inputgroup/index.css:48]: yep, you can take a look at https://www.chromatic.com/build?appId=5f0dd5ad2b5fc10022a2e320&number=77 (this is the run before the addition of `:not(.spectrum-InputGroup--quiet)`. At first I had `&.is-disabled:not(.spectrum-InputGroup--invalid)` only which broke quiet + read only comboboxes.

https://www.chromatic.com/build?appId=5f0dd5ad2b5fc10022a2e320&number=78 is the final run

Comment (snowystinger) [reply packages/@adobe/spectrum-css-temp/components/inputgroup/index.css:48]: great, thanks! i approved the tooltip ones, at some point need to figure out why those are flaky, i'll open an issue

**Review by snowystinger — APPROVED** [review]
lgtm

**Review by dannify — APPROVED** [review]

**Pull request merged**